index.js

```diff
@@ -4,59 +4,75 @@ const path = require('path');
 
 const run = (config, {logger}) => {
 
-	if(!Array.isArray(config.files)) {
-		return Promise.resolve({
-			status: 'error',
-			message: 'Configuration does not contain a valid "files" property.'
+    if(!Array.isArray(config.files)) {
+        return Promise.resolve({
+            status: 'error',
+            message: 'Configuration does not contain a valid "files" property.'
+        });
+    }
+
+    const promises = config.files.map(fileConfig => processFile(fileConfig, config.plugins, logger));
+
+    return Promise.all(promises)
+        .then(() => {
+            const message = `${config.files.length} stylesheet(s) processed.`;
+            logger.info(message);
+            return {
+                status: 'complete',
+                message: message
+            };
+        })
+        .catch(e => {
+            logger.error(e);
+            return {
+                status: 'error',
+                message: e.message
+            };
 		});
-	}
-
-	const promises = config.files.map(fileConfig => processFile(fileConfig, config.plugins, logger));
-
-	return Promise.all(promises)
-		.then(() => {
-			const message = `${config.files.length} stylesheet(s) processed.`;
-			logger.info(message);
-			return {
-				status: 'complete',
-				message: message
-			};
-		})
-		.catch(e => {
-			logger.error(e);
-			return {
-				status: 'error',
-				message: e.message
-			};
+	};
+
+const processFile = (fileConfig, plugins, logger) => {
+    const destinationArray = [];
+    const srcFilepath = path.resolve(process.cwd(), fileConfig.src);
+	handleDestinations(fileConfig, destinationArray);
```

Comment:
Instead of passing an empty array to the `handleDestinations` function and having that function modify the parameter, consider not passing the second parameter to the `handleDestinations` function and have it return an array.

So this line would become
```js
const destinationArray = getDestinations(fileConfit)
```

and the function would change to 
```js
const handleDestinations = (fileConfig) => {
    if (Array.isArray(fileConfig.dest)) {
        return [...fileConfig.dest];
    } else {
        return [fileConfig.dest];
    }
}
```